<launch>
  <!-- Name of overall setup -->
  <arg name="robot_name"                        default="testing_environment"/>

  <!-- Arm models/configs being used in the experimental setup -->
  <arg name="arm_A_robot_model"                 default="wx200"/>
  <arg name="arm_A_port"                        default="/dev/widow_200_arm_A"/>
  <arg name="arm_A_operating_mode"              default="position"/>
  <arg name="arm_B_robot_model"                 default="wx200"/>
  <arg name="arm_B_port"                        default="/dev/widow_200_arm_B"/>
  <arg name="arm_B_operating_mode"              default="position"/>

  <!-- Relative displacement between arm A and arm B -->
  <arg name="relative_x"                        default=""/>
  <arg name="relative_y"                        default=""/>
  <arg name="relative_z"                        default=""/>
  <arg name="relative_roll"                     default=""/>
  <arg name="relative_pitch"                    default=""/>
  <arg name="relative_yaw"                      default=""/>

  <!-- Perch x,y position and height relative to arm A base -->
  <arg name="perch_x"                           default=""/>
  <arg name="perch_y"                           default=""/>
  <arg name="perch_z"                           default=""/>

  <!-- Are we running this in Gazebo or the Real World? -->
  <arg name="use_gazebo"                         default="false"/>
  <arg name="use_actual"                         default="true"/>

  <!-- NEED TO EDIT THIS -->
  <!-- if we're running this in Gazebo -->
  <include if="$(arg use_gazebo)" file="$(find interbotix_gazebo)/launch/gazebo.launch">
    <arg name="robot_name"                        value="$(arg robot_name)"/>
  </include>


  <!-- NEED TO EDIT THIS -->
  <!-- if we're running this in the real world -->
  <include if="$(arg use_actual)" file="$(find testing_environment_moveit)/launch/multi_arm.launch">
    <arg name="robot_name"                        value="$(arg robot_name)"/>
    <arg name="arm_A_robot_model"                 value="$(arg arm_A_robot_model)"/>
    <arg name="arm_B_robot_model"                 value="$(arg arm_B_robot_model)"/>

    <arg name="arm_A_port"                        value="$(arg arm_A_port)"/>
    <arg name="arm_B_port"                        value="$(arg arm_B_port)"/>

    <arg name="arm_A_operating_mode"              value="$(arg arm_A_operating_mode)"/>
    <arg name="arm_B_operating_mode"              value="$(arg arm_B_operating_mode)"/>

    <arg name="relative_x"                        value="$(arg relative_x)"/>
    <arg name="relative_y"                        value="$(arg relative_y)"/>
    <arg name="relative_z"                        value="$(arg relative_z)"/>
    <arg name="relative_roll"                     value="$(arg relative_roll)"/>
    <arg name="relative_pitch"                    value="$(arg relative_pitch)"/>
    <arg name="relative_yaw"                      value="$(arg relative_yaw)"/>

    <arg name="perch_x"                           value="$(arg perch_x)"/>
    <arg name="perch_y"                           value="$(arg perch_y)"/>
    <arg name="perch_z"                           value="$(arg perch_z)"/>

    <arg name="use_default_rviz"                  value="false"/>
    <arg name="load_moveit_world_frame"           value="true"/>
  </include>

  <!-- NEED TO EDIT THIS -->
  <include file="$(find testing_environment_moveit)/launch/move_group.launch" ns="$(arg robot_name)">
    <arg name="publish_monitored_planning_scene" value="true" />
    <!-- <arg name="robot_name"                       value="$(arg robot_name)"/> -->
    <arg name="relative_x"                        value="$(arg relative_x)"/>
    <arg name="relative_y"                        value="$(arg relative_y)"/>
    <arg name="relative_z"                        value="$(arg relative_z)"/>
    <arg name="relative_roll"                     value="$(arg relative_roll)"/>
    <arg name="relative_pitch"                    value="$(arg relative_pitch)"/>
    <arg name="relative_yaw"                      value="$(arg relative_yaw)"/>
  </include>

  <!-- NEED TO EDIT THIS -->
  <include file="$(find interbotix_moveit)/launch/moveit_rviz.launch" ns="$(arg robot_name)">
    <arg name="config"                           value="true"/>
  </include>

</launch>
